servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultBlockingStreamingHttpServerResponse.java

```diff
@@ -132,8 +135,19 @@ public BlockingStreamingHttpServerResponse context(final ContextMap context) {
         return this;
     }
 
+    @Nonnull
+    @Override
+    public ContextMap context() {
+        final ContextMap ctx = super.context();
+        return isSent() ? new ReadOnlyContextMap(ctx) : ctx;
```

Comment:
> interesting part here are the failures where GRPC is using the response context for some trailer work

hm, I looked at gRPC `RequestResponseContextTest`. gRPC is a bit different. Context there is not part of the `BlockingStreamingGrpcServerResponse`, it's part of `GrpcServiceContext`. So, it's not obvious from the API that they are related to `response.sendMetaData()`. 

Then I checked how it's used in `ContextHttpServiceFilter` and it seems legit (thread safe) to me even if it's after `response.sendMetaData()`, because it happens before `writer.close()` and then it's read inside `ContextHttpServiceFilter` via `response.transform` operation, which guarantees that `payloadComplete` will be invoked after `setTrailersContext` (inside try-with-resources, like 372). 

That makes me feel that maybe we should not restrict writability of the response context. It should be fine as long as it's sequenced correctly. But I'm not sure how to make it easier to understand in javadoc of the response context.

Reply:
Ok, let's close this and reconsider a different approach w.r.t documenting it.